### PR TITLE
test: integration tests for null optional parameters

### PR DESCRIPTION
**Pull request by Yuan325**
Note: null optional parameters currently doesn't work with BigQuery.

### CODE REVIEW

File: tests/tool.go [hunk 1]
```diff
@@ -109,13 +109,21 @@ func RunToolInvokeTest(t *testing.T, select1Want, invokeParamWant, invokeParamWa
 			isErr:         false,
 		},
 		{
-			name:          "invoke my-param-tool2 with nil response",
-			api:           "http://127.0.0.1:5000/api/tool/my-param-tool2/invoke",
+			name:          "invoke my-param-tool-id with nil response",
```
Comment (duwenxin99) [inline tests/tool.go]: Renaming `my-param-tool-id` to `my-get-id-tool`, `my-param-tool-name` to `my-name-tool`?
Feeling like these test names get too long and don't make sense anymore...

File: tests/tool.go [hunk 2]
```diff
@@ -76,7 +76,7 @@ func RunToolGetTest(t *testing.T) {
 }
 
 // RunToolInvoke runs the tool invoke endpoint
-func RunToolInvokeTest(t *testing.T, select1Want, invokeParamWant, invokeParamWantNull string, supportsArray bool) {
+func RunToolInvokeTest(t *testing.T, select1Want, invokeParamWant, invokeIdParamWant, nullString string, supportNullParam, supportsArray bool) {
```
Comment (duwenxin99) [inline tests/tool.go]: If we are renaming `invokeParamWantNull` we should probably also rename it in each test files, otherwise the readability is decreased. Also, is it better to rename it "invokeIdWantNull" since it is testing null results as well? I think it is more descriptive, and also feeling the `Param` part isn't needed.

**Review by duwenxin99 — CHANGES_REQUESTED** [review]

File: tests/tool.go (hunk 1 above)
Comment (Yuan325) [reply tests/tool.go]: Updated `my-param-tool` --> `my-tool` , `my-param-tool-id` --> `my-tool-by-id`, `my-param-tool-name` --> `my-tool-by-name`. Since technically all 3 of them is accessing the same table. just with different parameters.

File: tests/tool.go (hunk 2 above)
Comment (Yuan325) [reply tests/tool.go]: Updated to `invokeIdNullWant` since `Want` is the suffix for the variable.
